workspaces/ui-v2/src/optic-components/hooks/config/AppConfiguration.tsx

```diff
@@ -0,0 +1,42 @@
+import React, { useContext } from 'react';
+
+interface IFeatureFlags {}
+
+interface IAppConfigurations {
+  navigation: {
+    showDocs?: boolean;
+    showChangelog?: boolean;
+    showDiff?: boolean;
+  };
+  documentation: {
+    allowDescriptionEditing: boolean;
+  };
+}
+
+export type OpticAppConfig = {
+  featureFlags: IFeatureFlags;
+  config: IAppConfigurations;
+};
+
+const AppConfigurationContext = React.createContext<OpticAppConfig | null>(
+  null
+);
+
+export const AppConfigurationStore = (props: {
+  config: OpticAppConfig;
+  children: any;
```

Comment:
React nodes can be typed as `React.ReactNode` - or just importing `import { ReactNode } from 'react'`

Or alternatively you could type `AppConfigurationStore` as a `React.FC` - with a prop input in the type - see https://github.com/opticdev/optic/pull/678/files#diff-3389b405e926d5b93ec3fde35293c19a3453b75ad659c8e7576e51bce0d95882R5-R9 for an example

Reply:
Good call -- I've updated these.